src/hotspot/share/interpreter/bytecodes.cpp

```diff
@@ -29,13 +29,309 @@
+  def(_newarray        , "newarray"        , "bc"   , nullptr    , T_OBJECT ,  0, true , _newarray       ) \
+  def(_anewarray       , "anewarray"       , "bkk"  , nullptr    , T_OBJECT ,  0, true , _anewarray      ) \
+  def(_arraylength     , "arraylength"     , "b"    , nullptr    , T_INT    ,  0, true , _arraylength    ) \
+  def(_athrow          , "athrow"          , "b"    , nullptr    , T_VOID   , -1, true , _athrow         ) \
+  def(_checkcast       , "checkcast"       , "bkk"  , nullptr    , T_OBJECT ,  0, true , _checkcast      ) \
+  def(_instanceof      , "instanceof"      , "bkk"  , nullptr    , T_INT    ,  0, true , _instanceof     ) \
+  def(_monitorenter    , "monitorenter"    , "b"    , nullptr    , T_VOID   , -1, true , _monitorenter   ) \
+  def(_monitorexit     , "monitorexit"     , "b"    , nullptr    , T_VOID   , -1, true , _monitorexit    ) \
+  def(_wide            , "wide"            , ""     , nullptr    , T_VOID   ,  0, false, _wide           ) \
+  def(_multianewarray  , "multianewarray"  , "bkkc" , nullptr    , T_OBJECT ,  1, true , _multianewarray ) \
+  def(_ifnull          , "ifnull"          , "boo"  , nullptr    , T_VOID   , -1, false, _ifnull         ) \
+  def(_ifnonnull       , "ifnonnull"       , "boo"  , nullptr    , T_VOID   , -1, false, _ifnonnull      ) \
+  def(_goto_w          , "goto_w"          , "boooo", nullptr    , T_VOID   ,  0, false, _goto_w         ) \
+  def(_jsr_w           , "jsr_w"           , "boooo", nullptr    , T_INT    ,  0, false, _jsr_w          ) \
+  def(_breakpoint      , "breakpoint"      , ""     , nullptr    , T_VOID   ,  0, true , _breakpoint     ) \
+  JVM_BYTECODES_DO(def)
+
+bool Bytecodes::_is_initialized = false;
+
+const char* const Bytecodes::_name[Bytecodes::number_of_codes] = {
+#define BYTECODE_NAME(code, name, format, wide_format, result_type, depth, can_trap, java_code) name,
+  BYTECODES_DO(BYTECODE_NAME)
+#undef BYTECODE_NAME
+};
+
+const BasicType Bytecodes::_result_type[Bytecodes::number_of_codes] = {
+#define BYTECODE_RESULT_TYPE(code, name, format, wide_format, result_type, depth, can_trap, java_code) result_type,
+  BYTECODES_DO(BYTECODE_RESULT_TYPE)
+#undef BYTECODE_RESULT_TYPE
+};
+
+const s_char Bytecodes::_depth[Bytecodes::number_of_codes] = {
+#define BYTECODE_DEPTH(code, name, format, wide_format, result_type, depth, can_trap, java_code) depth,
+  BYTECODES_DO(BYTECODE_DEPTH)
+#undef BYTECODE_DEPTH
+};
+
+// Helper for determining the size (a.k.a. length) of a string literal.
+struct StringLiteralSize {
+  template <size_t N>
+  static constexpr size_t invoke(const char (&)[N]) {
+    static_assert(N > 0, "N must be greater than 0");
+    // The size is N - 1, as C strings have an implicit NUL at the end. So "foo" will result in N
+    // being 4, but we actually want 3.
+    return N - 1;
+  }
+
+  static constexpr size_t invoke(std::nullptr_t) {
+    return 0;
+  }
+};
+
+#define STRING_SIZE(string) StringLiteralSize::invoke(string)
```

Comment:
Can't you simply use:
```
#define STRING_SIZE(string) (sizeof(string) - 1)
```
?

Reply:
No, due to `nullptr` for the `wide_format`. That was my first instinct as well, then I saw `nullptr`.

Reply:
Ah I see.